### PR TITLE
NIAD-1794: PoC for A tool requirements

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

File: vnp/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        inbound_queue
+
+
+This will send the content of the `example.xml` file to the `inbound_queue` on `localhost` broker listening on port `5672` connecting using unsecured AMQP protocol.
+
+There will be 3 messages each having different NHS number, because the source `example_nhs_numbers.txt` file has 3 lines each having different value.
+
+When connecting to AMQP broker using a secured connection, `amqps://` protocol should be used instead of `amqp://`
+
+Using https://stackoverflow.com/a/44078785/249136 for generating UUID in javascript
```
Comment (michael-kainos) [inline vnp/README.md:67]: Can we copy the example here incase this post is removed or archived

File: vnp/transform.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+msg.body=msg.body.replace('{NHS_NUMBER}', entry); 
+
```
Comment (michael-kainos) [inline vnp/transform.js:2]: What are the numbers for in this script? Can we use variables so their purpose is clearly understood?

File: vnp/README.md (hunk 1 above)
Comment (bartek-sarul) [reply vnp/README.md:67]: It's just a reference that the code for generating UUID was copied from that SO answer.
I mean these 2 lines of code:
https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/361/files#diff-1c6752bb6f54fd884d6ffc00ea4679e8422304b80e79202ee17dde7139585b80R3

I'll remove this link from this readme file and move it to the javascript as a comment

File: vnp/transform.js (hunk 2 above)
Comment (bartek-sarul) [reply vnp/transform.js:2]: The `transform.js` file is executed for every line of the [-W](https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/361/files#diff-b92f39385cdea983f292abb288f9aeb980befaa048abdff895908604eaa6f5a2R48) replacing the `{NHS_NUMBER}` placeholder with the `entry` variable that is filled in by the "A" tool by values from given file.
An example has been provided [here](https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/361/files#diff-48b3e573be33df3198bc213bd07a6cf170b1e25feebe4b7e57074b78a45df696)
And usage is [here](https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/361/files#diff-b92f39385cdea983f292abb288f9aeb980befaa048abdff895908604eaa6f5a2R52)

Comment (michael-kainos) [reply vnp/transform.js:5]: Sorry i may have not been clear, i am talking about the numbers in the UUID generate code

Comment (bartek-sarul) [reply vnp/transform.js:5]: those are just template parameters to properly format generated UUID

**Review by michael-kainos — APPROVED** [review]

**Review by ElliotB-96 — APPROVED** [review]

**Pull request merged**